### PR TITLE
feat: new permissions model

### LINKED ISSUE #102: Integrate new permissions model
Let's update asto and http and integrate new permissions model

### PR DESCRIPTION
**Pull request by olenagerasimova**
closes #102

### CODE REVIEW

File: src/main/java/com/artipie/conda/http/CondaSlice.java [hunk 1]
```diff
@@ -74,20 +76,21 @@ public String generate(final Authentication.User user) {
      * @param url Application url
      */
     public CondaSlice(final Storage storage, final String url) {
-        this(storage, Permissions.FREE, Authentication.ANONYMOUS, CondaSlice.ANONYMOUS, url);
+        this(storage, Policy.FREE, Authentication.ANONYMOUS, CondaSlice.ANONYMOUS, url, "test");
```
Comment (dgarus) [inline src/main/java/com/artipie/conda/http/CondaSlice.java:79]: Suggest to change `test` to `*`

File: src/main/java/com/artipie/conda/http/CondaSlice.java [hunk 2]
```diff
@@ -74,20 +76,21 @@ public String generate(final Authentication.User user) {
...
+     * @param policy Permissions
      * @param users Users
      * @param tokens Tokens
      * @param url Application url
+     * @param repo Repository name
      * @checkstyle ParameterNumberCheck (5 lines)
      */
-    public CondaSlice(final Storage storage, final Permissions perms, final Authentication users,
-        final Tokens tokens, final String url) {
+    public CondaSlice(final Storage storage, final Policy<?> policy, final Authentication users,
```
Comment (dgarus) [inline src/main/java/com/artipie/conda/http/CondaSlice.java:92]: As I understand, we have a separate slice for every repo. 
Is it true?

**Review by dgarus — CHANGES_REQUESTED** [review]
@olenagerasimova thanks!
Minor comment

File: src/main/java/com/artipie/conda/http/CondaSlice.java (hunk 2 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/conda/http/CondaSlice.java:92]: yes, there is an entry-point slice in each adapter. In artipie these adapter slices created in `SliceFromConfig` class

**Comment by olenagerasimova** [comment]
@dgarus thanks, fixed

**Review by dgarus — APPROVED** [review]
Thanks!

**Pull request merged**